service/collector/rate_limit.go

```diff
@@ -0,0 +1,262 @@
+package collector
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strconv"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
+	"github.com/Azure/go-autorest/autorest/to"
+	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/azure-operator/client"
+	"github.com/giantswarm/azure-operator/service/controller/v9/key"
+	"github.com/giantswarm/azure-operator/service/credential"
+)
+
+const (
+	remainingReadsHeaderName  = "x-ms-ratelimit-remaining-subscription-reads"
+	remainingWritesHeaderName = "x-ms-ratelimit-remaining-subscription-writes"
+)
+
+var (
+	ReadsDesc *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName("azure_operator", "RateLimit", "reads"),
```

Comment:
Shouldn't this be rather following?

```suggestion
		prometheus.BuildFQName("azure_operator", "rate_limit", "reads"),
```